F/FLINT/build_tarballs.jl

```diff
@@ -45,4 +41,16 @@ dependencies = [
 ]
 
 # Build the tarballs, and possibly a `build.jl` as well.
-build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies)
+build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies;
+               init_block = """
+  if !Sys.iswindows() && !__isthreaded[]
+    #to match the global gmp ones
+    fm = dlsym(libflint_handle, :__flint_set_memory_functions)
+    ccall(fm, Nothing,
+      (Ptr{Nothing},Ptr{Nothing},Ptr{Nothing},Ptr{Nothing}),
+        cglobal(:jl_malloc),
+        cglobal(:jl_calloc),
+        cglobal(:jl_realloc),
+        cglobal(:jl_free))
+  end
+""")
```

Comment:
With this in, assuming it actually works as intended (I have not yet checked), we could probably get rid of LoadFlint.jl -- at least once we drop support for Julia 1.0-1.2 in our packages (e.g. if/when Julia 1.6 is released as a new LTS).

Reply:
```suggestion
               init_block = """
  if !Sys.iswindows() && !(get(ENV, "NEMO_THREADED", "") == "1")
    #to match the global gmp ones
    fm = dlsym(libflint_handle, :__flint_set_memory_functions)
    ccall(fm, Nothing,
      (Ptr{Nothing},Ptr{Nothing},Ptr{Nothing},Ptr{Nothing}),
        cglobal(:jl_malloc),
        cglobal(:jl_calloc),
        cglobal(:jl_realloc),
        cglobal(:jl_free))
  end
""")
```